projects/ngx-avatar/src/lib/sources/initials.ts

```diff
@@ -10,15 +10,15 @@ export class Initials implements Source {
 
   constructor(public sourceId: string) {}
 
-  public getAvatar(initialsSize: number): string {
-    return this.getInitials(this.sourceId, initialsSize);
+  public getAvatar(size: number): string | null {
+    return this.getInitials(this.sourceId, size);
   }
 
   /**
    * Returns the initial letters of a name in a string.
    */
   private getInitials(name: string, size: number): string {
-    name = name ? name.trim() : null;
+    name = name ? name.trim() : '';
 
     if (!name) {
```

Comment:
Changing the minimum as possible.

```suggestion
    if (name === '') {
```

Reply:
I did not touch this line. I am not sur what you mean with this suggestion. Are you saying that all empty string check should be explicit in the entire project ?

Reply:
Nope, I'm just following my previous suggestion here, no criticism to your changes at all. I think this is a good opportunity to improve this byte of code.

Reply:
Latest code, after tslint fixing, looks like the following. Are you sure you still want to change the second `if (!name) {` into `if (name === '') {` ?

```ts
  private getInitials(name: string, size: number): string {
    if (!name) {
      return '';
    }

    name = name.trim();

    if (!name) {
      return '';
    }

    const initials = name.split(' ');
```

Reply:
Yes, why check for `name` if you already checked it. The only reason for this check is because `.trim()` can make a `'   '` string become a `''` string. Since JS is a strange place I gotta ask: is there any reason why this wouldn't work?

Reply:
What I wrote above (A)

or B:

```ts
  private getInitials(name: string, size: number): string {
    name = name.trim();

    if (!name) {
      return '';
    }

    const initials = name.split(' ');
```

or C:


```ts
  private getInitials(name: string, size: number): string {
    if (!name) {
      return '';
    }

    name = name.trim();

    if (name === '') {
      return '';
    }

    const initials = name.split(' ');
```

would all work. Which one is your favorite ?

Reply:
If strings could be trusted I would say B, but as they can't, I like C because of what I already said in here. Can we trust that with strict mode enabled no null values will be passed down to the `name` param?

Reply:
Strict mode is only enabled for our own code, at compilation time. So we indeed cannot trust that somebody will mis-use this lib. But should we support misusage ?

For this PR I assumed that types declared on inputs of `AvatarComponent` are correct and respected by end-users. Those types are also documented as non-null in README. If we choose to support nullable things, then we should be explicit about it and modify most/all inputs in `AvatarComponent`, as well as the documentation in README.

My opinion is to keep existing types (minimal change for us), support only non-null and defer the responsibility to end-user, which should not be that big of a problem.

Reply:
Then B. Let me know if you'd like to change or I can merge.

Reply:
done